cmd/clusterctl/client/repository/repository_github_test.go

```diff
@@ -543,7 +566,7 @@ func Test_gitHubRepository_getLatestRelease(t *testing.T) {
 		{
 			name: "Get latest release, ignores pre-release version",
 			field: field{
-				providerConfig: config.NewProvider("test", "https://github.com/o/r1/releases/latest/path", clusterctlv1.CoreProviderType),
+				providerConfig: config.NewProvider("test", "https://github.com/o/r1/releases/v0.4.2/path", clusterctlv1.CoreProviderType),
```

Comment:
I'm not sure about these changes replacing latest with a specific release, because the goal of this test is to validare how latest is resolved

Reply:
This does not change calling `latestRelease` which tests the actual behaviour.

This only changes that `NewGithubRepository` (which has separate test cases) does not call implicitly `latestContractRelease` which errors out with the new behaviour because of no latestRelease is found.

For the cases 1 and 3, we could also fake that api calls to succeed and proceed with the test.

For 2: this would always fail because `NewGithubRepository` calls `latestRelease` implicitly.

Reply:
thanks for clarification